servant/src/Servant/API/Raw.hs

```diff
@@ -11,4 +12,5 @@ import Data.Typeable (Typeable)
 -- In addition to just letting you plug in your existing WAI 'Application's,
 -- this can also be used with 'Servant.Utils.StaticFiles.serveDirectory' to serve
 -- static files stored in a particular directory on your filesystem
-data Raw deriving Typeable
+newtype Raw a (m :: * -> *) = Raw a
```

Comment:
I'd write out all the instances `Identity` (from [Data.Functor.Identity](https://hackage.haskell.org/package/base-4.8.0.0/docs/Data-Functor-Identity.html)) here. And maybe also have an `unRaw` record.

I think the documentation should say something like 

> the `m` type parameter is a phantom type used to allow <http://hackage.haskell.org/package/servant-server/docs/Servant-Server.html#g:4 `enter`> to treat endpoints uniformly. 

We're going to need a better explanation for that, but I haven't really figured out how best to phrase it...